src/includes.ts

```diff
@@ -0,0 +1,63 @@
+import { find, pipe } from ".";
+import IterableInfer from "./types/IterableInfer";
+import ReturnValueType from "./types/ReturnValueType";
+import { isAsyncIterable, isIterable } from "./_internal/utils";
+
+/**
+ * Checks if the specified value is equal. (contains)
+ *
+ * @example
+ * ```ts
+ * includes('c', 'abcd'); // true
+ * includes('e', 'abcd'); // false
+ * includes(3, [1,2,3,4]); // true
+ * includes(5, [1,2,3,4]); // false
+ * ```
+ */
+function includes<T extends readonly []>(value: unknown, iterable: T): false;
+
+// prettier-ignore
+function includes<T>(
+    value: unknown, 
+    iterable: Iterable<T>
+): boolean;
+
+function includes<T>(
+  value: unknown,
+  iterable: AsyncIterable<T>,
+): Promise<boolean>;
+
+function includes<T extends Iterable<unknown> | AsyncIterable<unknown>>(
+  value: unknown,
+): (iterable: T) => ReturnValueType<T, boolean>;
+
+function includes<T extends Iterable<unknown> | AsyncIterable<unknown>>(
+  value: unknown,
+  iterable?: T,
+): boolean | Promise<boolean> | ((iterable: T) => ReturnValueType<T, boolean>) {
+  if (iterable === undefined) {
+    return (iterable) => {
+      return includes(value, iterable as any) as ReturnValueType<T, boolean>;
+    };
+  }
+
+  if (isIterable(iterable)) {
+    return pipe(
+      iterable as Iterable<IterableInfer<T>>,
+      find((a) => a === value),
+      (a) => a !== undefined,
```

Comment:
How about implementing using `some` like this?

```suggestion
    return some(a => a === value, iterable);
```

Reply:
oh yes your right :)